lib/measure_repository_service_test_kit/library_group.rb

```diff
@@ -47,5 +51,145 @@ class LibraryGroup < Inferno::TestGroup
+      id 'read-and-search-library-05'
+      description %(This test verifies that a Library resource can be found through search by identifier from
+      the server.)
+      input :library_identifier, title: 'Library Identifier'
+
+      run do
+        fhir_search(:library, params: { identifier: library_identifier })
+
+        assert_response_status(200)
+        assert_resource_type(:bundle)
+        assert_valid_json(response[:body])
+        assert(!resource.entry[0].nil?, 'Search by identifier returned an empty FHIR searchset bundle')
+        assert resource_has_matching_identifier?(resource.entry[0].resource, library_identifier),
+               "Requested resource with identifier #{library_identifier}, received resource with identifier
+        #{resource.entry[0].resource.identifier}"
+      end
+    end
+
+    test do
+      title 'Server returns 200 response status and bundle that contains all the correct verisons of the library
+      matching a name'
+      id 'read-and-search-library-06'
+      description %(This test verifies that a Library resource can be found through search by name from the server.)
+      input :library_name, title: 'Library name'
+
+      run do
+        fhir_search(:library, params: { name: library_name })
+
+        assert_response_status(200)
+        assert_resource_type(:bundle)
+        assert_valid_json(response[:body])
+        assert(!resource.entry[0].nil?, 'Search by name returned an empty FHIR searchset bundle')
+        assert resource.entry[0].resource.name.downcase.include?(library_name.downcase), "Requested resource
+        with name #{library_name}, received resource with name #{resource.entry[0].resource.name}"
+      end
+    end
+
+    test do
+      title 'Server returns 200 response status and bundle that contains all the correct versions of the library
+      matching a title'
+      id 'read-and-search-library-07'
+      description %(This test verifies a Library resource can be found through search by title from the server.)
+      input :library_title, title: 'Library title'
+
+      run do
+        fhir_search(:library, params: { title: library_title })
+
+        assert_response_status(200)
+        assert_resource_type(:bundle)
+        assert_valid_json(response[:body])
+        assert(!resource.entry[0].nil?, 'Search by title returned an empty FHIR searchset bundle')
+        assert resource.entry[0].resource.title.downcase.include?(library_title.downcase), "Requested resource
+        with title #{library_title}, received resource with title #{resource.entry[0].resource.title}"
+      end
+    end
+
+    test do
+      title 'Server returns 200 response status and bundle that contains all the correct versions of the library
+      matching a status'
+      id 'read-and-search-library-08'
+      description %(This test verifies a Library resource can be found through search by status from the server.)
+      input :library_status, title: 'Library status'
```

Comment:
I would update this input to a radio, since there are only 4 valid options: `draft | active | retired | unknown` same for Measure